lib/ecto/changeset.ex

```diff
@@ -1984,6 +1985,23 @@ defmodule Ecto.Changeset do
     true
   end
 
+  defp ensure_field_not_many!(%Changeset{types: types}, field) do
```

Comment:
We can merge this function with `ensure_fields_exists!`, so we don't lookup `types` twice. :)

Reply:
Actually `ensure_fields_exists!` doesn't get called, because we filter out by `missing?` first before calling `ensure_fields_exists!`.
`missing?` calls `get_field`, gets `[]`, and return `false`.
Will can still avoid looking it up twice and actually look it up before looping on `fields`.
Pushed https://github.com/elixir-ecto/ecto/pull/3862/commits/245b4e179f11acf386fdf947d19584993772b070